bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAsyncTable.java

```diff
@@ -87,6 +96,26 @@ private static <T, R> List<R> map(List<T> list, Function<T, R> f) {
   private final TableName tableName;
   private BatchExecutor batchExecutor;
 
+  private static final Boolean MUTATE_ROW_RETURNS_RESULT;
+
+  static {
+    Type wrappedReturnType = null;
+    try {
+      Method mutateRow = Table.class.getDeclaredMethod("mutateRow", RowMutations.class);
+      ParameterizedType returnType = (ParameterizedType) mutateRow.getGenericReturnType();
+      wrappedReturnType = returnType.getActualTypeArguments()[0];
+    } catch (NoSuchMethodException e) {
+    }
+
+    if (wrappedReturnType == Void.class) {
+      MUTATE_ROW_RETURNS_RESULT = false;
+    } else if (wrappedReturnType == Result.class) {
+      MUTATE_ROW_RETURNS_RESULT = true;
+    } else {
+      MUTATE_ROW_RETURNS_RESULT = null;
```

Comment:
what will happen in this case? should we throw exception instead?

Reply:
updated it to throw an exception at class instantiation time